src/main/scala/gitbucket/core/service/SystemSettingsService.scala

```diff
@@ -296,14 +348,34 @@ object SystemSettingsService {
     password: Option[String]
   )
 
-  case class SshAddress(host: String, port: Int, genericUser: String)
+  case class SshAddress(host: String, port: Int, genericUser: String) {
+
+    def isDefaultPort: Boolean =
+      port == PublicSshPort
+
+    def getUrl: String =
+      if (isDefaultPort) {
+        s"${genericUser}@${host}"
+      } else {
+        s"${genericUser}@${host}:${port}"
+      }
+
+    def getUrl(owner: String, name: String): String =
+      if (isDefaultPort) {
+        s"${genericUser}@${host}:${owner}/${name}.git"
```

Comment:
Isn't `ssh://` necessary?

Reply:
It isn’t when the port is 22! The only time it’s required is if the port is different.

Reply:
Oh, I see. Thank you for the clarification.